### PR TITLE
add workflow for building RevealJS slides

### PR DESCRIPTION
**Pull request by millhaus**
adds:
- build-slides.yaml: builds RevealJS and deploys to staging
- push-main-slides.yaml builds and deploys to prod
- pr-cleanup-slides.yaml removes staging deployment
- yaml files for `kubectl create -f` deployment, ingress and service on both prod & test on Rancher

### CODE REVIEW

File: yaml/slides_prod/slides_ingress_prod.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  annotations:
+    field.cattle.io/publicEndpoints: '[{"addresses":["10.100.1.8"],"port":443,"protocol":"HTTPS","serviceName":"pitc-ansible-training-port:ansible-slides","ingressName":"pitc-ansible-training-prod:ansible-slides","hostname":"ansible-slides.k8s.puzzle.ch","path";"/","allNodes":true}]'
```
Comment (ombre8) [inline yaml/slides_prod/slides_ingress_prod.yaml:5]: do we need those annotations and internal ip here?

File: yaml/slides_test/slides_ingress_test.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  annotations:
+    field.cattle.io/publicEndpoints: '[{"addresses":["10.100.1.8"],"port":443,"protocol":"HTTPS","serviceName":"pitc-ansible-training-test:ansible-slides","ingressName":"pitc-ansible-training-test:ansible-slides","hostname":"ansible-slides-test.k8s.puzzle.ch","path";"/","allNodes":true}]'
```
Comment (ombre8) [inline yaml/slides_test/slides_ingress_test.yaml:5]: same as above

**Review by ombre8 — CHANGES_REQUESTED** [review]

File: yaml/slides_prod/slides_ingress_prod.yaml (hunk 1 above)
Comment (millhaus) [reply yaml/slides_prod/slides_ingress_prod.yaml]: Don't think so. Remove it

File: yaml/slides_test/slides_ingress_test.yaml (hunk 2 above)
Comment (millhaus) [reply yaml/slides_test/slides_ingress_test.yaml]: same as above

**Review by ombre8 — APPROVED** [review]
lgtm, thx

**Pull request merged**